e2e-tests/src/elections.rs

```diff
@@ -0,0 +1,82 @@
+use std::{collections::HashSet, iter::empty};
+
+use aleph_client::{get_validators_for_session, AccountId, AnyConnection};
+use log::debug;
+use primitives::{CommitteeSeats, EraValidators, SessionIndex};
+
+pub fn get_members_for_session<C: AnyConnection>(
+    connection: &C,
+    seats: CommitteeSeats,
+    era_validators: &EraValidators<AccountId>,
+    session: SessionIndex,
+) -> (Vec<AccountId>, Vec<AccountId>) {
+    let reserved_members_for_session =
+        get_members_subset_for_session(seats.reserved_seats, &era_validators.reserved, session);
+    let non_reserved_members_for_session = get_members_subset_for_session(
+        seats.non_reserved_seats,
+        &era_validators.non_reserved,
+        session,
+    );
+
+    let reserved_members_bench =
+        get_bench_members(&era_validators.reserved, &reserved_members_for_session);
+    let non_reserved_members_bench = get_bench_members(
+        &era_validators.non_reserved,
+        &non_reserved_members_for_session,
+    );
+    let members_bench = empty()
+        .chain(reserved_members_bench)
+        .chain(non_reserved_members_bench)
+        .collect();
+
+    let members_active: Vec<_> = empty()
+        .chain(reserved_members_for_session)
+        .chain(non_reserved_members_for_session)
+        .collect();
+
+    let members_active_set: HashSet<_> = members_active.iter().cloned().collect();
+    let network_members: HashSet<_> = get_validators_for_session(connection, session)
+        .into_iter()
+        .collect();
+
+    debug!(
+        "expected era validators for session {}: reserved - {:?}, non-reserved - {:?}",
+        session, era_validators.reserved, era_validators.non_reserved
+    );
+    debug!("Seats for session {}: {:?}", session, seats);
+    debug!(
+        "members for session - computed {:?} ; retrieved {:?}",
+        members_active, network_members
+    );
+
+    assert_eq!(members_active_set, network_members);
+
+    (members_active, members_bench)
+}
+
+pub fn get_members_subset_for_session(
+    nodes_per_session: u32,
+    era_validators: &[AccountId],
+    session: SessionIndex,
+) -> Vec<AccountId> {
+    let validators_len = era_validators.len();
+    let session: usize = session.try_into().unwrap();
+    let nodes_per_session: usize = nodes_per_session.try_into().unwrap();
```

Comment:
just `into()`?

Reply:
I don't see how. My `u32` doesn't implement it.